test-data/unit/lib-stub/enum.pyi

```diff
@@ -37,4 +41,7 @@ class IntFlag(int, Flag):
 
 
 class auto(IntFlag):
-    value: Any
\ No newline at end of file
+
+    value: Any
+
+    def __init__(self) -> None: pass
```

Comment:
I don't know why I need to add this.  It _isn't_ present in `typeshed`, but my actual runs still seem to work.  Without it, I have two tests that fail with complaints that `auto` didn't get the right number of arguments:

```
[case testEnumValueSomeAuto]
from enum import Enum, auto
class Truth(Enum):
    true = 8675309
    false = auto()

def infer_truth(truth: Truth) -> None:
    reveal_type(truth.value) # N: Revealed type is 'builtins.int'
```

And

```
[case testEnumValueCustomAuto]
from enum import Enum, auto
from typing import List, Any
class AutoName(Enum):

    # In `typeshed`, this is a staticmethod and has more arguments,
    # but I have lied a bit to keep the test stubs lean.
    def _generate_next_value_(self) -> str:
        return "name"

class Truth(AutoName):
    true = auto()
    false = auto()

def infer_truth(truth: Truth) -> None:
    reveal_type(truth.value) # N: Revealed type is 'builtins.str'
```

If anyone can help me reason through why _that_ would be, I'm happy to dig deeper.

Reply:
I had the same thought, but as far as I can tell, the same thing happens in the official stubs in [`typeshed`](https://github.com/python/typeshed/blob/master/stdlib/3/enum.pyi) -- but we don't see this complaint when I run it for real.  It also only complains for these two tests (which admittedly _are_ doing something a little special with `auto`) -- even though there are a lot of other tests that use `auto`.  I've never really seen `mypy` come up with a different required signature for something based on surrounding context.  Maybe there is something interesting happening in the plugin that I haven't quite grokked yet -- particularly when mixed with the slightly less-than-usual stub setup for these tests?

Reply:
Adding `[builtins fixtures/primitives.pyi]` to those two tests resolves this. I think I have an idea: without that, you get `int` and `object` from the *very* basic `lib-stub/builtins.pyi`, and there `int` has no `__init__`. But in `fixtures/primitives.pyi` there is an `int.__init__`.

The secret is in the MRO of `auto`, which is (in either case) `auto`, `IntFlag`, `int`, `Flag`, `Enum`, `object`. With `primitives.pyi`, we pick up `__init__` from `int`. But without it, we pick it up from `Enum`... And `Enum.__init__` requires an argument.

So it's purely a test issue, and IMO the fix is to add `[builtins fixtures/primitives.pyi]` and get rid of `auto.__init__`. (And of the blank line before `value: Any`.)